### PR TITLE
multi repo handling when opening error from codestream

### PR DESCRIPTION
**Pull request by eric-j**

### CODE REVIEW

File: shared/agent/src/providers/newrelic.ts [hunk 1]
```diff
@@ -3691,6 +3686,35 @@ export class NewRelicProvider extends ThirdPartyIssueProviderBase<CSNewRelicProv
 		return undefined;
 	}
 
+	private findRelatedReposFromServiceEntity(
```
Comment (bcanzanella) [inline shared/agent/src/providers/newrelic.ts:3689]: could this be merged with `findBuiltFrom()`?

File: shared/ui/Stream/CodeError/RepositoryAssociator.tsx [hunk 2]
```diff
@@ -99,22 +105,46 @@ export function RepositoryAssociator(props: {
...
-
-				setOpenRepositories(results);
+				//take repos in users IDE, and filter them with a list of
+				//related repos to service entity the error originates from
+				const filteredResults = results.filter(_ => {
+					return derivedState.relatedRepos.some(repo => {
+						return repo.name === _.name;
+					});
+				});
+				if (filteredResults.length < 2) {
```
Comment (bcanzanella) [inline shared/ui/Stream/CodeError/RepositoryAssociator.tsx:115]: will we ever see an empty array here?

File: shared/agent/src/providers/newrelic.ts [hunk 3]
```diff
@@ -621,11 +621,11 @@ export class NewRelicProvider extends ThirdPartyIssueProviderBase<CSNewRelicProv
 			const metricResponse = await this.getMetricData(request.errorGroupGuid);
 			if (!metricResponse) return undefined;
 
-			const mappedEntity = await this.findMappedRemoteByEntity(metricResponse?.entityGuid);
+			const mappedRepoEntities = await this.findMappedRemoteByEntity(metricResponse?.entityGuid);
 			return {
 				entityId: metricResponse?.entityGuid,
 				occurrenceId: metricResponse?.traceId,
-				remote: mappedEntity?.url,
+				relatedRepos: mappedRepoEntities,
```
Comment (bcanzanella) [inline shared/agent/src/providers/newrelic.ts:628]: just confirming that the `remote` property is no longer needed.

File: shared/ui/Stream/ObservabilityAssignmentsDropdown.tsx [hunk 4]
```diff
@@ -84,7 +84,11 @@ export const ObservabilityAssignmentsDropdown = React.memo((props: Props) => {
 												if (response) {
 													dispatch(
 														openErrorGroup(_.errorGroupGuid, response.occurrenceId, {
-															remote: response.remote,
+															multipleRepos: response?.relatedRepos
+																? response.relatedRepos.length > 1
```
Comment (bcanzanella) [inline shared/ui/Stream/ObservabilityAssignmentsDropdown.tsx:88]: could this be simplified to `response?.relatedRepos?.length > 1`

File: shared/ui/Stream/ObservabilityErrorDropdown.tsx [hunk 5]
```diff
@@ -81,18 +81,39 @@ export const ObservabilityErrorDropdown = React.memo((props: Props) => {
...
+											onClick={async e => {
+												try {
+													const response = (await HostApi.instance.send(
+														GetObservabilityErrorGroupMetadataRequestType,
+														{ errorGroupGuid: err.errorGroupGuid }
+													)) as GetObservabilityErrorGroupMetadataResponse;
+													dispatch(
+														openErrorGroup(err.errorGroupGuid, err.occurrenceId, {
+															multipleRepos: response?.relatedRepos
+																? response.relatedRepos.length > 1
```
Comment (bcanzanella) [inline shared/ui/Stream/ObservabilityErrorDropdown.tsx:93]: same comment as above

File: shared/ui/Stream/ObservabilityErrorDropdown.tsx [hunk 6]
```diff
@@ -81,18 +81,39 @@ export const ObservabilityErrorDropdown = React.memo((props: Props) => {
...
+																: undefined,
+															remote: response?.remote
+																? response.remote
+																: observabilityRepo.repoRemote,
+															timestamp: err.lastOccurrence,
+															sessionStart: derivedState.sessionStart,
+															pendingEntityId: response?.entityId
+																? response.entityId
+																: err.entityId,
+															occurrenceId: response?.occurrenceId
```
Comment (bcanzanella) [inline shared/ui/Stream/ObservabilityErrorDropdown.tsx:106]: i believe these could be coalesced like `response?.occurrenceId || err.occurrenceId`

File: shared/agent/src/providers/newrelic.ts (hunk 3 above)
Comment (eric-j) [reply shared/agent/src/providers/newrelic.ts:628]: correct, when fetching metaData for the error, we just want the relatedRepos array now

**Review by bcanzanella — APPROVED** [review]

**Pull request merged**